src/Command/CheckUpdateCommand.php

```diff
@@ -59,27 +69,32 @@ protected function configure() : void
 
     protected function execute(InputInterface $input, OutputInterface $output) : void
     {
-        $loggerHelper = new LoggerHelper();
-        $logger = $loggerHelper->create($input->getOption('debug'));
+        $logger = $this->getLogger($input);
 
         $logger->debug('started checking for new version of remote file');
 
-        $browscap = new BrowscapUpdater();
-
-        $browscap->setLogger($logger);
-        $browscap->setCache($this->getCache($input));
+        $browscap = new BrowscapUpdater($this->getCache($input), $logger);
         $browscap->checkUpdate();
 
         $logger->debug('finished checking for new version of remote file');
     }
 
-    private function getCache(InputInterface $input) : BrowscapCacheInterface
+    private function getCache(InputInterface $input) : CacheInterface
     {
         if (null === $this->cache) {
-            $cacheAdapter = new File([File::DIR => $input->getOption('cache')]);
-            $this->cache = new BrowscapCache($cacheAdapter);
+            $fileCache = new FilesystemCache($input->getOption('cache'));
+            $this->cache = new SimpleCacheAdapter($fileCache);
         }
 
         return $this->cache;
     }
+
+    private function getLogger(InputInterface $input) : LoggerInterface
+    {
+        if (null === $this->logger) {
+            $this->logger = LoggerHelper::createDefaultLogger($input->getOption('debug'));
```

Comment:
Let's get rid of this `getThing` pattern where we silently instantiate deps statically. If a dep is required, create it in the constructor so the object is already in valid state (unless there's a good reason not to). Even better, make it a non-optional dependency, and leave the DI wiring to figure out what to inject

Reply:
(same comment applies for the rest of the commands btw, to avoid duplication)

Reply:
done